ci/Dockerfile.fedora

```diff
@@ -43,13 +43,21 @@ ARG NCPU=2
 ARG CXX=g++
 ARG CC=gcc
 ARG BUILD_TYPE=Debug
+ARG CHECK_STYLE=""
 
 # We assume that this is running on a (clean-ish) checkout of
 # google-cloud-cpp, including submodules, and copy the files to a
 # directory where we will build.
+WORKDIR /var/tmp/build
 COPY . /var/tmp/build/gccpp
+RUN test "${CHECK_STYLE}" != "yes" || cp -r gccpp gccpp-orig
 
 WORKDIR /var/tmp/build/gccpp/build-output
 RUN cmake -DCMAKE_BUILD_TYPE=${BUILD_TYPE} ..
 RUN make -j ${NCPU} all
 RUN make -j ${NCPU} test || ( cat Testing/Temporary/LastTest.log; exit 1 )
+
+RUN test "${CHECK_STYLE}" != "yes" || make clang-format
```

Comment:
I'm a bit confused... this is running in the `build-output` tree, no? Should it run in `gccpp` tree instead?

Would it be easier if this were done in a shell script rather than via Dockerfile commands (and could be shared across docker files)?

Reply:
You run in the `build-output` tree because that is where the Makefiles are generated.  They point to the source in `gccpp`.

The version of `clang-format` can (and does!) change your output, so we want a super controlled environment to run it, so this must be a script running inside the Docker container, but yes.  It would have to be something like:

```
RUN ci/check-style.sh
```

I can change the CL to that if you want.

Reply:
Up to you; it would be nice to share code (looks like it's duplicated verbatim across two Dockerfiles), but you can do that in this PR or a follow-up cleanup PR.

Reply:
Done.  It was a much better idea than I thought, the glob expression in `bigtable/CMakeList.txt` was broken and missing some files.

Reply:
Great! Is there a target / command that developers can run to reformat their code? Something in the CMakeLists or the generated Makefile? Or do they just run `clang-format -i bigtable/` and done? This should also be documented in the README as it's now a required part of the PR, since the test will fail if they don't do this, and once it does, they won't know how to fix it.

Reply:
Updated the README.md file.